lib/github-service.js

```diff
@@ -29,6 +37,23 @@ class GithubService {
     });
   }
 
+  async createPullRequestComment(report){
+    getAssociatedPullRequestNumberFromSha(this.getOctokitInstance(), this.options.repo, this.options.refCommit)
+    .then((response) => {
+      if (response.data.length == 0){
```

Comment:
The method createPullRequestComment needs to return a Promise in both case (length = 0 or not)
Because this is being dealt with in Promise.allSettled (expecting a promise to be returned by the methods in the array)

As well, It's better to have the nominal case first since this is what matter the most when you're reading the code, and then alternative, error cases. (so length check at the end)

 ```
if (response.data.length) {
  return this.getOctokitInstance().request('POST /repos/{owner}/{repo}/issues/{issue_number}/comments', {
          owner: this.options.repo.split('/')[0]...
        ....
}
return Promise.reject(new Error('Not pull request found for commit id  : " + this.options.refCommit))
```

Reply:
Done